docs/source/groupby-api.rst

```diff
@@ -0,0 +1,19 @@
+API
```

Comment:
I have a slight preference to keep all the api docs for each module together. Would you be opposed to moving this into the `dataframe-api.rst` file?